ddtrace/internal/_encoding.pyx

```diff
@@ -264,17 +441,23 @@ cdef class MsgpackEncoderBase(BufferedEncoder):
     cpdef flush(self):
         raise NotImplementedError()
 
-    cdef pack_span(self, object span, char *dd_origin):
+    cdef pack_span(self, object span, void *dd_origin):
         raise NotImplementedError()
 
 
-cdef class MsgpackEncoder(MsgpackEncoderBase):
+cdef class MsgpackEncoderV03(MsgpackEncoderBase):
     cpdef flush(self):
         try:
             return self.get_bytes()
         finally:
             self._reset_buffer()
 
+    cdef void * get_dd_origin_ref(self, object dd_origin):        
+        IF PY_MAJOR_VERSION >= 3:
+            return <void *> PyUnicode_AsUTF8(dd_origin)
+        ELSE:
+            return <void *> (<char *> dd_origin)
```

Comment:
What a beauty 🤩

Reply:
aside/suggestion: we should add a common helper somewhere to handle `PyObject` -> `char*` and the other way around.

I ran into some issues trying to get bytes/str/unicode to consistently go to char* and back without issues. Probably user error for sure, but made me think having a single helper/converter would be a nice to have.